test/unit/utils-service-test.js

```diff
@@ -0,0 +1,176 @@
+describe("UNIT SERVICE - utils", function () {
+    "use strict";
+
+    // Angular injectables
+    var $q, $httpBackend;
+
+    // Module defined (non-Angular) injectables
+    var utils, config, utilsMockResponse;
+
+    // Initialize modules
+    beforeEach(function () {
+
+        module("TendrlModule");
+        module("TestDataModule");
+    });
+
+    beforeEach(function () {
+        inject(function (_$httpBackend_, _$q_, _config_, _utilsMockResponse_) {
+            $httpBackend = _$httpBackend_;
+            $q = _$q_;
+            config = _config_;
+            utilsMockResponse = _utilsMockResponse_;
+        });
+
+        inject(function (_utils_) {
+            utils = _utils_;
+        });
+
+    });
+
+    it("Should get desired action details from getActionDetails", function() {
+        
+         // Setup data
+        config.baseUrl = "https://abc:8080",
+        config.clusterId = "1234";
+
+        utils.setActionDetails({}, "Create");
+
+        // Exercise SUT
+        expect(utils.getActionDetails()).to.deep.equal({action: {method: "POST", url: "https://abc:8080cluster/1234/volume/create"}, actionName: "Create"});
+    });
+
+    it("Should perform action when form is submitted", function() {
+        var data, response;
+
+        // Setup data
+        data = {
+                Node :["279-78774-782"],
+                sds_name:"sds",
+                sds_version:"sds"
+            };
+
+        response = {
+            "job_id": "1234",
+            "job_status": "progress"
+        };
+
+        // Setup data - expectation
+        $httpBackend.expectPOST("https//abc.com")
+            .respond(200, response);
+
+        // Exercise SUT
+        utils.takeAction(data, "https//abc.com")
+            .then(function (data) {
+            // Verify result (state)
+            expect(data.job_id).to.equal(response.job_id);
+            expect(data.job_status).to.equal(response.job_status);
+        });
+        $httpBackend.flush();
+
+
+        // Verify result (behavior)
+        $httpBackend.verifyNoOutstandingExpectation();
+        $httpBackend.verifyNoOutstandingRequest();
+    });
+
+    it("Should get list of options whe type is 'LIST'", function() {
```

Comment:
only text changes require - " Should get list of options where type is 'LIST'  "